src/Modules/Visualization/ShowField.cc

```diff
@@ -448,754 +538,293 @@ void GeometryBuilder::renderFacesLinear(
-  std::vector<Vector> vvals;
-  std::vector<Tensor> tvals;
-  std::vector<ColorRGB> face_colors;
 
-  if (fld->basis_order() < 0 || state.get(RenderState::USE_DEFAULT_COLOR))
-  {
-    colorScheme = ColorScheme::COLOR_UNIFORM;
-  }
-  else if (state.get(RenderState::USE_COLORMAP))
+  auto realColorMap = colorMap.get();
+  if(!realColorMap) realColorMap = StandardColorMapFactory::create();
+
+  auto map = StandardColorMapFactory::create(realColorMap->getColorMapName(),
+    realColorMap->getColorMapResolution(), realColorMap->getColorMapShift(),
+    realColorMap->getColorMapInvert());
+
+  auto coordinateMap = StandardColorMapFactory::create("Grayscale", 256, 0, false,
+    realColorMap->getColorMapRescaleScale(), realColorMap->getColorMapRescaleShift());
+
+  if (useColorMap)
   {
-    numAttributes += 4;
+    numAttributes += 2;
     colorScheme = ColorScheme::COLOR_MAP;
   }
-  else // if (fld->basis_order() >= 0)
-  {
-    numAttributes += 4;
-    colorScheme = ColorScheme::COLOR_IN_SITU;
-  }
 
   Point idpt;
   VMesh::Face::iterator fiter, fiterEnd;
   VMesh::Node::array_type nodes;
-  mesh->begin(fiter);
-  mesh->end(fiterEnd);
   mesh->get_nodes(nodes, *fiter);
   mesh->get_point(idpt, nodes[0]);
+  mesh->begin(fiter);
+  mesh->end(fiterEnd);
 
-  // Three 32 bit ints for each triangle to index into the VBO (triangles = verticies - 2)
-  size_t iboSize = static_cast<size_t>(mesh->num_faces() * sizeof(uint32_t) * (nodes.size() - 2) * 3);
-  size_t vboSize = static_cast<size_t>(mesh->num_faces() * sizeof(float) * nodes.size() * numAttributes);
-
-  // Construct VBO and IBO that will be used to render the faces. Once again, IBOs are not strictly
-  // needed. But, we may be able to optimize this code somewhat.
-  // todo Switch to unique_ptrs and move semantics.
-  std::shared_ptr<spire::VarBuffer> iboBufferSPtr(new spire::VarBuffer(iboSize));
-  std::shared_ptr<spire::VarBuffer> vboBufferSPtr(new spire::VarBuffer(vboSize));
+  int numNodes = nodes.size();
+  bool useQuads = (numNodes == 4);
+  int writeCase = ((int)useQuads << 2) | ((int)useNormals << 1) | ((int)useColorMap);
 
-  // Accessing the pointers like this is contrived. We only do this for speed since we will be
-  // using the pointers in a tight inner loop.
-  auto iboBuffer = iboBufferSPtr.get();
-  auto vboBuffer = vboBufferSPtr.get();
+  std::vector<Point> points(numNodes);
+  std::vector<Vector> normals(numNodes);
+  std::vector<glm::vec2> textureCoords(numNodes);
+  std::vector<double> svals(numNodes);
+  std::vector<Vector> vvals(numNodes);
+  std::vector<Tensor> tvals(numNodes);
 
-  uint32_t iboIndex = 0;
-  size_t numVBOElements = 0;
+  size_t passNumber = 0;
+  size_t facesLeft = mesh->num_faces();
 
-  while (fiter != fiterEnd)
+  while(facesLeft > 0)
   {
-    interruptible->checkForInterruption();
+    const static size_t maxFacesPerPass = 1 << 24;
+    int facesLeftInThisPass = std::min(facesLeft,maxFacesPerPass);
+    facesLeft -= facesLeftInThisPass;
```

Comment:
This logic should be extracted into a `partitionFaces` function, which then can pass the correct subset of faces to the actual renderFaces call. It will most likely change in the future

Reply:
I don't follow

Reply:
The structure is 
```while (stuffToDo())
{ doStuff(); }
```
I'm saying extract each of those two parts as a separate function since they are independent.

Reply:
Independent from each other, and reusable for nodes/edges partitioning.